### PR TITLE
fix #26 bug 1202082: skip publicsuffix domains

### LINKED ISSUE #26: Filter out public suffixes from the blacklist
As an extra precaution, we should ensure that blacklists don't contain any bare [public suffixes](https://publicsuffix.org/). For example, we don't want to start blocking `.com` if that gets onto the list by mistake.

Something like `.com` is obvious and would most likely get caught by our manual review process, but others aren't. `googleapis.com` is currently [on the list](https://github.com/mozilla-services/shavar-prod-lists/blob/master/disconnect-blacklist.json#L7486) and that's [preventing resources from being whitelisted](https://bugzilla.mozilla.org/show_bug.cgi?id=1202082).

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

File: lists2safebrowsing.py [hunk 1]
```diff
@@ -200,6 +206,11 @@ def find_hosts(disconnect_json, allow_list, chunk, output_file, log_file,
             d = d.encode('utf-8');
             canon_d = canonicalize(d);
             if (not canon_d in domain_dict) and (not d in allow_list):
+              # check if the domain is in the public suffix list
+              # SafeBrowsing keeps trailing '/', PublicSuffix does not
+              psl_d = canon_d.rstrip('/')
+              if psl.publicsuffix(psl_d) == psl_d:
+                continue
```
Comment (fmarier) [inline lists2safebrowsing.py]: It would be nice if we could log these.

**Comment by groovecoder** [comment]
@fmarier - I added a line to log when PSL domains are skipped. You can see it if you do this:

```
$ ./lists2safebrowsing.py
...
$ ag "Public" *.log
content-track-digest256.log:35:[Public Suffix] cloudfront.net; Skipping.
content-track-digest256.log:1020:[Public Suffix] googleapis.com; Skipping.

mozfull-track-digest256.log:156:[Public Suffix] cloudfront.net; Skipping.
mozfull-track-digest256.log:1141:[Public Suffix] googleapis.com; Skipping.

mozfullstaging-track-digest256.log:159:[Public Suffix] cloudfront.net; Skipping.
mozfullstaging-track-digest256.log:1144:[Public Suffix] googleapis.com; Skipping.
```

**Comment by groovecoder** [comment]
@fmarier how does this look? Good to merge?

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,3 @@
 boto==2.40.0
+publicsuffixlist==0.4.0
+requests==2.10.0
```
Comment (fmarier) [inline requirements.txt]: Are you using `requests` anywhere? I don't see it in the diff...

**Comment by fmarier** [comment]
>  how does this look? Good to merge?

Can you also check that the number of entries (based on the log files) in each list is the same before and after your change?

File: requirements.txt (hunk 2 above)
Comment (groovecoder) [inline requirements.txt]: I'm not, but the `publicsuffixlist` module is even though it doesn't have its own `requirements.txt` nor `setup.py` to require it. So I had to include it here myself. 😢

**Comment by groovecoder** [comment]
Left is this branch; right is master.

tl;dr - the # of items are identical except `tracking-protection-base`, `tracking-protection-content`, `tracking-protection-full`, `staging-tracking-protection-full` each have 2 fewer items. (i.e., `cloudfront.net` and `googleapis.com`)

psl-vs-master-list-items

**Comment by fmarier** [comment]
Looks great! r+
